students/wooseok-choi/user/views.py

```diff
@@ -1 +1,30 @@
+import json
 
+from django.http  import JsonResponse
+from django.views import View
+
+from user.models import User
+from user.validate import *
+
+
+class SignupView(View):
+    def post(self, request):
+        data = json.loads(request.body)
+        
+        try:
+            if not validate_email(data['email']):
+                return JsonResponse({'message':'INVALID EMAIL'}, status=400)
+            if not validate_password(data['password']):
+                return JsonResponse({'message':'INVALID PASSWORD'}, status=400) 
+            if not duplicate_email(data['email']):
+                return JsonResponse({'message':'DUPLICATE EMAIL'}, status=409)
```

Comment:
```suggestion
            if User.objects.filter(email=data['email'].exists:
                return JsonResponse({'message':'DUPLICATE EMAIL'}, status=409)
```
해당 방법을 추천 드립니다. 왜 이 방법이 더 좋을까 한 번 생각해 보시면 좋을 것 같아요

Reply:
제가 처음에 작성한 try문은 email이 존재 유무 뿐만 아니라 다른 에러들도 예외처리가 나와서 명확하지 않은 함수였던 것 같아요
또 함수에 try문을 넣고 그 함수를 import 하는 것은 비효율적인 것 같습니다. 쿼리셋의 exists 메소드를 사용하는 것이 훨씬 명확하고 간결한 코드인 것 같아요!